batch-nft-reveal/app/components/collection/Mint.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  Box,
+  Button,
+  Container,
+  Heading,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  Text,
+} from '@chakra-ui/react'
+import { useState } from 'react'
+import { useContractFunction, useEthers } from '@usedapp/core'
+import { Contract } from '@ethersproject/contracts'
+import { useContractCall } from '../../hooks/useContractCall'
+import { providers, utils, BigNumber } from 'ethers'
+import json from '../../artifacts/contracts/NFTCollection.sol/NFTCollection.json'
+import { AddressProp } from '../../types/AddressProp'
+import { Error } from '../Error'
+
+export const Mint = (props: AddressProp): JSX.Element => {
+  let contract: Contract
+
+  const { account } = useEthers()
+  const { contractAddress } = props
+  const { abi } = json
+
+  const name: string = useContractCall('name', [], contractAddress)
+  const symbol: string = useContractCall('symbol', [], contractAddress)
+  const totalSupply: BigNumber = useContractCall(
+    'totalSupply',
+    [],
+    contractAddress
+  )
+  const costToMint: BigNumber = useContractCall('mintCost', [], contractAddress)
+
+  const [mintAmount, setMintAmount] = useState(1)
+  const [isMintDisabled, setIsMintDisabled] = useState(false)
+
+  if (contractAddress) {
+    contract = new Contract(
+      contractAddress,
+      new utils.Interface(abi),
+      providers.getDefaultProvider()
+    )
+  }
+
+  const { send, state } = useContractFunction(contract, 'mint', {})
+  const isLoading = state.status === 'Mining'
+
+  return (
+    <Container maxWidth="100%" centerContent>
+      <Heading>
+        {name}({symbol})
```

Comment:
It would be nice to have placeholders before the data is pulled.
 
<img width="665" alt="Screenshot 2022-08-24 at 17 18 50 (2)" src="https://user-images.githubusercontent.com/482351/186442657-82c64ea7-a94d-42fb-afc6-d28c546a87f0.png">


See example: https://keepers.chain.link/rinkeby/42657775885172339278827540325965760402148640157821653590619741824028976585602